pkg/security/certManager/client/certManager.go

```diff
@@ -7,18 +7,26 @@ import (
 	"github.com/plgd-dev/hub/v2/pkg/fsnotify"
 	"github.com/plgd-dev/hub/v2/pkg/log"
 	"github.com/plgd-dev/hub/v2/pkg/security/certManager/general"
+	"github.com/plgd-dev/hub/v2/pkg/strings"
 )
 
-// Config provides configuration of a file based Server Certificate manager
+// Config provides configuration of a file based Server Certificate manager. CAPool can be a string or an array of strings.
 type Config struct {
-	CAPool          string `yaml:"caPool" json:"caPool" description:"file path to the root certificates in PEM format"`
-	KeyFile         string `yaml:"keyFile" json:"keyFile" description:"file name of private key in PEM format"`
-	CertFile        string `yaml:"certFile" json:"certFile" description:"file name of certificate in PEM format"`
-	UseSystemCAPool bool   `yaml:"useSystemCAPool" json:"useSystemCaPool" description:"use system certification pool"`
+	CAPool          interface{} `yaml:"caPool" json:"caPool" description:"file path to the root certificates in PEM format"`
+	KeyFile         string      `yaml:"keyFile" json:"keyFile" description:"file name of private key in PEM format"`
+	CertFile        string      `yaml:"certFile" json:"certFile" description:"file name of certificate in PEM format"`
+	UseSystemCAPool bool        `yaml:"useSystemCAPool" json:"useSystemCaPool" description:"use system certification pool"`
+	CAPoolArray     []string    `yaml:"-" json:"-"`
```

Comment:
I don't like that this is exported, this is an internal field generated from the `CAPool` field.

What about something like this:

```go
type Config struct {
...
  caPoolArray []string `yaml:"-" json:"-"`...
  validated bool
}

func (c *Config) CAPoolArray()([]string, error) {
  if !c.validated {
    return nil, error
  }
  return c.caPoolArray, nil
}
```

Reply:
fixed

Reply:
yep, good job.